iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java

```diff
@@ -345,6 +347,22 @@ public EntrypointTree ENTRYPOINT() {
     );
   }
 
+  public HealthCheckTree HEALTHCHECK() {
+    return b.<HealthCheckTree>nonterminal(DockerLexicalGrammar.HEALTHCHECK).is(
+      b.firstOf(
+        f.healthcheck(
+          b.token(DockerKeyword.HEALTHCHECK),
+          b.token(DockerKeyword.NONE)
+        ),
+        f.healthcheck(
+          b.token(DockerKeyword.HEALTHCHECK),
+          b.zeroOrMore(PARAM()),
+          CMD()
```

Comment:
You could also consider NONE as an instruction, like CMD. Then the second parameter would simply be an instruction tree. You would make the grammar narrower and do without the second constructor. Also, you don't have the problem of redundant fields `none` and `cmd`. The only disadvantage would be, you must create an instruction `NONE`, which may appear only in HEALTHCHECK. IMO this solution would be less hacky.

Reply:
Very interesting approach ! I will give it a try, sounds good to me 👍

Reply:
Using this approach, it is possible to parse such instruction : `HEALTHCHECK --interval=30s NONE`
I was thinking it was invalid for docker, but apparently it is not ! Two birds with one stone 😶‍